### PR TITLE
Present an empty list if list is expected but no data

### LINKED ISSUE #47: Inconsistent behavior of present when no rows are found
If the result of a DBIx::Class query given to `present` returns no rows, the generated JSON is `undef`, rather than an empty array. Can the behaviour be changed to make it return `[]`, so as to make the behaviour consistent?

Many thanks,
Dave

### PR DESCRIPTION
**Pull request by khrt**
Fixes #47

### CODE REVIEW

File: lib/Raisin/Entity.pm [hunk 1]
```diff
@@ -69,6 +69,8 @@ sub compile {
         while (my $i = $data->next) {
             push @$result, _compile_column($entity, $i, \@expose);
         }
+
+        $result = [] unless $result;
     }
     # Array
     elsif (ref($data) eq 'ARRAY') {
```
Comment (davel) [inline lib/Raisin/Entity.pm]: You might want to default `$result` to `[]` for the case where `ref($data) eq 'ARRAY'` as well.

Comment (khrt) [reply lib/Raisin/Entity.pm]: Isn't `$result` equals to `[]` if `ref($data) eq 'ARRAY'` returns true?

Comment (davel) [reply lib/Raisin/Entity.pm]: It doesn't look like it. If we have `$data = []`, `$result` will remain `undef`. `$result` only gets a value inside the loop, on lines 76/78, below.

Comment (khrt) [reply lib/Raisin/Entity.pm]: Correct. Thanks for pointing out to that!

**Pull request merged**